Unable to encode Value type in a container

Easiest to illustrate with an example, attempting to encode an empty `Vec<Value>`

```rust
use bendy::{value::Value,encoding::ToBencode};

fn main() {
    let _ = Vec::<Value>::new().to_bencode();
}
```

Results in a compile error:

```
error[E0080]: could not evaluate constant
   --> $path/bendy-0.3.0/src/encoding/to_bencode.rs:114:38
    |
114 |             const MAX_DEPTH: usize = ContentT::MAX_DEPTH + 1;
    |                                      ^^^^^^^^^^^^^^^^^^^^^^^ attempt to add with overflow
```

Which I suspect is due to `Value::MAX_DEPTH` being the max_usize. A saturating add would be ideal in this case, but using a slightly lower `MAX_DEPTH` with room for a bunch parent containers may work, too.

--- a/src/value.rs
+++ b/src/value.rs
@@ -60,7 +60,9 @@ impl<'a> Value<'a> {
 }
 
 impl<'a> ToBencode for Value<'a> {
-    const MAX_DEPTH: usize = usize::max_value();
+    // This leaves some room for external containers.
+    // TODO(#38): Change this to 0 for v0.4
+    const MAX_DEPTH: usize = usize::max_value() / 4;
 
     fn encode(&self, encoder: SingleItemEncoder) -> Result<(), crate::encoding::Error> {
         match self {
@@ -73,6 +75,8 @@ impl<'a> ToBencode for Value<'a> {
 }
 
 impl<'a> FromBencode for Value<'a> {
+    const EXPECTED_RECURSION_DEPTH: usize = <Self as ToBencode>::MAX_DEPTH;
+    
     fn decode_bencode_object(object: Object) -> Result<Self, crate::decoding::Error> {
         match object {
             Object::Bytes(bytes) => Ok(Value::Bytes(Cow::Owned(bytes.to_owned()))),

